<?xml version='1.0' encoding='utf-8'?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro" name="maci">

    <!-- Include a UR10 -->
    <xacro:include filename="$(find ur_description)/urdf/ur_macro.xacro" />
    <xacro:arg name="ur_type" default="ur5e"/>

    <xacro:ur_robot name="ur5" tf_prefix="ur5_" parent="world" joint_limits_parameters_file="$(find ur_description)/config/$(arg ur_type)/joint_limits.yaml" kinematics_parameters_file="$(find ur_description)/config/$(arg ur_type)/default_kinematics.yaml" physical_parameters_file="$(find ur_description)/config/$(arg ur_type)/physical_parameters.yaml" visual_parameters_file="$(find ur_description)/config/$(arg ur_type)/visual_parameters.yaml" sim_ignition="true" initial_positions="${dict(shoulder_pan_joint=-0.15,shoulder_lift_joint=0,elbow_joint=0.0,wrist_1_joint=0,wrist_2_joint=0.0,wrist_3_joint=0.0)}">
        <origin xyz="-0.1 0 0.15" rpy="${pi} 0 0" />
    </xacro:ur_robot>
    <link name="world" />

    <xacro:include filename="$(find robotiq_2f_140_gripper_visualization)/urdf/robotiq_arg2f_140_model.xacro" />
    <joint name="gripper_connect" type="fixed">
        <parent link="ur5_tool0" />
        <child link="robotiq_arg2f_base_link" />
    </joint>

    <xacro:macro name="ros2_position_controller" params="joint_name">
        <joint name="${joint_name}">
            <command_interface name="position"/>
            <state_interface name="position">
                <param name="initial_value">0.0</param>
            </state_interface>
            <state_interface name="velocity"/>
        </joint>
    </xacro:macro>

    <ros2_control name="ur5_" type="system">
        <hardware>
            <plugin>ign_ros2_control/IgnitionSystem</plugin>
        </hardware>

        <xacro:ros2_position_controller joint_name="finger_joint" />
        <xacro:ros2_position_controller joint_name="left_inner_finger_joint" />
        <xacro:ros2_position_controller joint_name="right_outer_knuckle_joint" />
        <xacro:ros2_position_controller joint_name="right_inner_finger_joint" />
    </ros2_control>

    <gazebo>
        <plugin filename="ign_ros2_control-system" name="ign_ros2_control::IgnitionROS2ControlPlugin">
            <parameters>$(find maci)/config/ros2_control.yaml</parameters>
        </plugin>
    </gazebo>

</robot>